### DISCUSSION TITLE
Ímage conversion issue

### DISCUSSION THREAD

**Discussion by amannagarro01** [ImageMagick/ImageMagick#7719] [General]
I have few .tiff and .ai format files, for which ImageMagick convert command is unable to generate thumbnails/previews.
Can anyone help here. Attaching files
[Test Images.zip](https://github.com/user-attachments/files/17506627/Test.Images.zip)

Commands used-
magick convert "20090721 602 (1).tif" -strip outputai800.png

**Comment by dlemstra** [comment]
Can you please describe in more detail what you mean by "conversion issue"? This will help other people who run into something similar in the future and also make it easier for us to figure out what kind of help you need.

**Reply by fmw42** [reply]
On  IM 7, use magick, not magick convert.

**Reply by amannagarro01** [reply]
I am trying to generate thumbnail of attached .tiff/.ai file in .png format so that they can be use for preview in our DAM(Digital Access Management) webportal.
For generating thumbnails we use ImageMagick convert command, example of which is mentioned in the above description.

**Reply by dlemstra** [reply]
Could you also answer my question?
> Can you please describe in more detail what you mean by "conversion issue"?

**Reply by amannagarro01** [reply]
@fmw42 , I have tried magick command as well still same issue.

**Reply by amannagarro01** [reply]
@dlemstra, by conversion issue I mean converting original image to .png thumbnail using ImageMagick convert command.

**Reply by dlemstra** [reply]
What is the issue?? what happens?

**Comment by snibgo** [comment]
Using the tif file in the linked zip, this command ...
```
magick "20090721 602 (1).tif" x.png
```
... raises these errors:
```
magick: 20090721 602 (1).tif: Null count for "Tag 34022" (type 1, writecount -3, passcount 1). `_TIFFVSetField' @error/tiff.c/TIFFErrors/592.
magick: 20090721 602 (1).tif: Null count for "Tag 34025" (type 1, writecount -3, passcount 1). `_TIFFVSetField' @error/tiff.c/TIFFErrors/592.
magick: 20090721 602 (1).tif: Null count for "Tag 34026" (type 1, writecount -3, passcount 1). `_TIFFVSetField' @error/tiff.c/TIFFErrors/592.
```
We can tell IM to ignore those tags:
```
magick -define tiff:ignore-tags=34022,34025,34026 "20090721 602 (1).tif" out.png
```
That command succeeds.

**Reply by amannagarro01** [reply]
Thanks @snibgo, I checked this command worked for me as well.
But the problem is we need are generating previews for thousands of file in one go from codebase by creating ImageMagick commands. For few such file how can we ignore tags dynamically, because attached file is just an example, there are several such files. It seems some problem in metadata of these files stopping them from generating previews.

**Comment by snibgo** [comment]
Ideally, go back to the source of those files to fix the metadata problem.

If the problems are always in those tag numbers, then you might always ignore them.

For more flexibility, a shell script can run magick without ignoring any tags. If there is an error, parse stderr to build a list of tag numbers that have problems. Then re-run magick, ignoring those tags.
